src/service/triple-store-service.js

```diff
@@ -71,38 +70,76 @@ class TripleStoreService {
         //     knowledgeAssetsUALs.map((ual) => `<${ual}>`),
         // );
         // const unifiedGraphTriples = [...triples, ...tripleAnnotations];
+        const publicKnowledgeAssetsTriples = this.dataService.groupTriplesBySubject(
+            triples.public ?? triples,
+        );
 
         const promises = [];
+        if (triples.private && triples.private.length !== 0 && !existsInNamedGraphs) {
```

Comment:
Perhaps we want separate queries to checki existence of public and private named graphs?